core/app.py

```
@@ -144,8 +171,24 @@ def fire_event(self, event_name: str, *args):
     # ─── Incoming Data Handlers ─────────────────────
 
     def _on_data(self, text: str):
-        self.trigger_manager.check_triggers(text)
-        self.main_window.console.echo(text)
+        """
+           Process incoming data from the MUD, optionally suppressing output for tracked mute patterns.
+           """
+        muted_patterns = []
+        for pattern in self._mute_patterns.keys():
```

Ruff (0.12.2)

Use `key in dict` instead of `key in dict.keys()`

Remove `.keys()`

(SIM118)